File: Samples/Verbum/how-to-enable-line-numbering-and-count-the-document-rows-t531470-20.1.2-/VB/LineNumberingExample/MyLayoutVisitor.vb
Imports DevExpress.XtraRichEdit.API.Layout
Imports DevExpress.XtraRichEdit.API.Native
Imports System

Namespace LineNumberingExample
	#Region "#MyLayoutVisitor"
	Public Class MyLayoutVisitor
		Inherits LayoutVisitor

		Private document As Document

		Private privateRowIndex As Integer
		Public Property RowIndex() As Integer
			Get
				Return privateRowIndex
			End Get
			Private Set(ByVal value As Integer)
				privateRowIndex = value
			End Set
		End Property

		Public Sub New(ByVal doc As Document)
			Me.document = doc
			RowIndex = 0
		End Sub

		Protected Overrides Sub VisitRow(ByVal row As LayoutRow)
			RowIndex += 1
			MyBase.VisitRow(row)
		End Sub
	End Class
	#End Region ' #MyLayoutVisitor
End Namespace

File: Samples/Verbum/how-to-enable-line-numbering-and-count-the-document-rows-t531470-20.1.2-/CS/LineNumberingExample/MyLayoutVisitor.cs
using DevExpress.XtraRichEdit.API.Layout;
using DevExpress.XtraRichEdit.API.Native;
using System;

namespace LineNumberingExample {
    #region #MyLayoutVisitor
    public class MyLayoutVisitor : LayoutVisitor {
        Document document;

        public int RowIndex { get; private set; }

        public MyLayoutVisitor(Document doc) {
            this.document = doc;
            RowIndex = 0;
        }

        protected override void VisitRow(LayoutRow row) {
            RowIndex++;
            base.VisitRow(row);
        }
    }
    #endregion #MyLayoutVisitor
}